src/notation/internal/notationinteraction.cpp

```diff
@@ -735,21 +735,25 @@ void NotationInteraction::drag(const PointF& fromPos, const PointF& toPos, DragM
     m_dragData.ed.moveDelta = delta - m_dragData.elementOffset;
     m_dragData.ed.evtDelta = evtDelta;
     m_dragData.ed.pos     = toPos;
+    m_dragData.ed.modifiers = QGuiApplication::keyboardModifiers();
 
-    if (isTextEditingStarted()) {
-        m_textEditData.pos = toPos;
-        toTextBase(m_textEditData.element)->dragTo(m_textEditData);
+    if (isTextEditStarted()) {
+        m_editData.pos = toPos;
+        toTextBase(m_editData.element)->dragTo(m_editData);
 
         notifyAboutTextEditingChanged();
         return;
     }
 
     if (isGripEditStarted()) {
-        m_dragData.ed.curGrip = m_gripEditData.curGrip;
+        m_dragData.ed.curGrip = m_editData.curGrip;
         m_dragData.ed.delta = m_dragData.ed.pos - m_dragData.ed.lastPos;
         m_dragData.ed.moveDelta = m_dragData.ed.delta - m_dragData.elementOffset;
-        m_dragData.ed.addData(m_gripEditData.getData(m_gripEditData.element));
-        m_gripEditData.element->editDrag(m_dragData.ed);
+        m_dragData.ed.addData(m_editData.getData(m_editData.element));
+        m_editData.element->editDrag(m_dragData.ed);
+    } else if (isElementEditStarted()) {
+        m_dragData.ed.delta = evtDelta;
```

Comment:
weird, but the base implementation for Element::editDrag (sorry, EngravingItem::editDrag) requires this (for dragging in edit mode)